W-17292458 MQ FIFO Exchanges Beta (#178)
mulesoft/docs-mq: modules/ROOT/pages/mq-apis.adoc

}, 
----

[[fifo-exchange-api]]
=== Create a FIFO Exchange

You can create a FIFO exchange using the Anypoint MQ Admin API by including `"fifo": true` in the body.

[NOTE]
The organization from which you use the Administration portal must have an Anypoint MQ FIFO entitlement.
See xref:mq-faq.adoc[Anypoint MQ FAQ] for the regions where Anypoint MQ is available.

Use a `curl` command like the following to create a FIFO exchange:

[source,bash,linenums]
----
curl -X PUT "https://anypoint.mulesoft.com/mq/admin/api/v1/organizations/<ORG_ID>/environments/<ENV_ID>/regions/<REGION_ID>/destinations/exchanges/<EXCHANGE_ID>" \
--header 'Content-Type: application/json' \
--header 'Authorization: bearer <BEARER_TOKEN>' \
--data-raw '{
  "encrypted" : false,
  "fifo" : true
}'
----

The response includes output such as:

[source,json,linenums]
----

> <NEED EXAMPLE OUTPUT>

Author: @kchoudhari-mule - we need example output. Also, is one example enough? It seems like we might want more here...

}, 
----

[[fifo-exchange-api]]
=== Create a FIFO Exchange

You can create a FIFO exchange using the Anypoint MQ Admin API by including `"fifo": true` in the body.

[NOTE]
The organization from which you use the Administration portal must have an Anypoint MQ FIFO entitlement.
See xref:mq-faq.adoc[Anypoint MQ FAQ] for the regions where Anypoint MQ is available.

> Use a `curl` command like the following to create a FIFO exchange:

Author: ```suggestion
Use a `curl` command like this to create a FIFO exchange:
```
